frontend/src/containers/LoginPage/LoginPage.js

```diff
@@ -1,31 +1,97 @@
 import React from 'react';
 import { Grid, Hidden } from '@material-ui/core';
 import LoginForm from '../../components/LoginForm/LoginForm';
+import { authenticate, getCurrentUser } from '../../api/PublicAPI';
+import CustomSnackbar from '../../components/CustomSnackbar/CustomSnackbar';
 import { videoTimeDrawing } from '../../assets';
 import './styles.css';
 
 class LoginPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showWrongCredentialsError: false,
+      showGeneralError: false,
+    };
+  }
+
+  handleLogin = (mail, password) => {
+    authenticate(mail, password)
+      .then((responseToken) => {
+        const token = responseToken.data;
+        getCurrentUser(token)
+          .then((responseUserData) => {
+            const { id, firstName, lastName, email } = responseUserData.data;
+            sessionStorage.setItem('token', token);
+            sessionStorage.setItem('id', id);
+            sessionStorage.setItem('firstName', firstName);
+            sessionStorage.setItem('lastName', lastName);
+            sessionStorage.setItem('email', email);
+            window.location.reload();
+          })
+          .catch(() => this.setState({ showGeneralError: true }));
+      })
+      .catch((ex) => {
+        try {
```

Comment:
You don't need a try-catch here. Just check if response is null/undefined or not

Reply:
done